File: C_Programming/Practise_Programs/PP_111.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_111
//
//  Description       : This program displays each character of a
//                      string using a character pointer and while loop.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 13/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

int main()
{
    char *str = "Ganesh";      

    while(*str != '\0')
    {
        printf("%c\n", *str);
        str++;
    }    

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      G
//      a
//      n
//      e
//      s
//      h
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_111.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_111
//
//  Description       : This program accepts a string from the user,
//                      displays the length of the string, and then
//                      displays each character of the string.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 26/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class PP_111
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        String Arr = null;

        System.out.println("Enter String : ");
        Arr = sobj.nextLine();

        System.out.println("Length of String is : " + Arr.length());

        int i = 0;

        for(i = 0; i < Arr.length(); i++)
        {
            System.out.println(Arr.charAt(i));
        }

        sobj.close();
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Ganesh
//
//  Sample Output :
//
//      Enter String :
//      Length of String is : 6
//      G
//      a
//      n
//      e
//      s
//      h
//
//////////////////////////////////////////////////////////////////